src/Network/Transport/TCP.hs

```diff
@@ -517,20 +519,25 @@ data TransportInternals = TransportInternals
 --------------------------------------------------------------------------------
 
 -- | Create a TCP transport
-createTransport :: N.HostName
-                -> N.ServiceName
+createTransport :: N.HostName    -- ^ Bind host name.
+                -> N.ServiceName -- ^ Bind port.
+                -> (N.ServiceName -> (N.HostName, N.ServiceName))
+                   -- ^ External address host name and port, computed from the
+                   --   actual bind port (must be lazy in it).
```

Comment:
> The haddock still asks for laziness

Will patch that.

> how the arrangement in the last patch has any effect on evaluation.

Right, `transport` is not forced so I suppose it doesn't matter.